auth/store.go

```diff
@@ -1284,7 +1284,7 @@ func NewTokenProvider(
 }
 
 func (as *authStore) WithRoot(ctx context.Context) context.Context {
-	if !as.IsAuthEnabled() {
+	if tt := tokenProviderType(as); tt != "simple" || !as.IsAuthEnabled() {
```

Comment:
we probably should define "simple" as a const.

Reply:
@xiang90 good idea, done.